modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -681,6 +703,9 @@ protected void mergeOptions(FieldMapper other, List<String> conflicts) {
                 this.shingleFields[i] = (ShingleFieldMapper) this.shingleFields[i].merge(m.shingleFields[i]);
             }
         }
+        if (Objects.equals(this.fieldType().similarity(), other.fieldType().similarity()) == false) {
+            conflicts.add("mapper [" + name() + "] has different [similarity] settings");
```

Comment:
was this also thrown before?

Reply:
As above, it was silently ignored.  I'll open an issue to make this emit a warning in 7x